website/source/assets/javascripts/configuration-builder.js

```diff
@@ -0,0 +1,119 @@
+function downloadConfiguration() {
+  var form = document.querySelector("#configuration-builder");
+  var config = "";
+
+  // Add Listener stanza
+  if (document.getElementById("include_tcp_listener").checked) {
+    config += 'listener "tcp" {\n' + addFieldsToStanza("listener") + '}\n';
+  }
+
+  // Add Storage stanza
+  if (document.getElementById("include_storage").checked) {
+    var backend = document.getElementById("storage").value;
+    config += '\nstorage "${backend}" {\n' + addFieldsToStanza("storage") + '}\n';
+  }
+
+  // Add Telemetry stanza
+  if (document.getElementById("include_telemetry").checked) {
+    var provider = document.getElementById("telemetry").value;
+    config += '\nstorage "${backend}" {\n' + addFieldsToStanza("storage") + '}\n';
+  }
+
+  // Add Seal stanza
+  if (document.getElementById("include_seal").checked) {
+    var type = document.getElementById("seal").value;
+    config += '\nstorage "${backend}" {\n' + addFieldsToStanza("seal") + '}\n';
+  }
+
+  // Add UI stanza
+  if (document.getElementById("include_ui").checked &&
+    document.getElementById("ui").value == true) {
+    config += '\nui = true';
+    var startServerLink = document.querySelector(".start-server-link");
+    startServerLink.href = startServerLink.href + "?tab=ui";
+  }
+
+  config = config.replace(/([^\r])\n/g, "$1\r\n");
+  var blob = new Blob([config], {type: "text/plain;charset=utf-8"});
+  saveAs(blob, "vault-config.hcl");
+  document.querySelector(".form-actions").style.display = "none";
+  document.querySelector("#download-confirm").style.display = "block";
+}
+
+function addFieldsToStanza(stanza) {
+  var fieldsets = document.querySelectorAll('[data-config-stanza="' + stanza + '"] .nested-fields fieldset');
+  var lines = "";
+
+  for (var i = 0; i < fieldsets.length; i++) {
+    var fieldset = fieldsets[i];
+    if (fieldset.offsetWidth > 0 && fieldset.offsetHeight > 0) {
+      var line = fieldsetToLine(fieldset);
+      if (line) {
+        lines += line;
+      }
+    }
+  }
+  return lines;
+}
+
+function fieldsetToLine(fieldset) {
+  var parameter = fieldset.getAttribute("name");
+  var isChecked = document.querySelector("#include_" + parameter).checked;
+  if (isChecked) {
+    var field = fieldset.querySelector("#" + parameter);
+    var value = field.value;
+
+    if (field.getAttribute("type") == "number") {
+      return '  ' + parameter + ' = ' + value + '\n';
+    } else {
+      return '  ' + parameter + ' = "' + value + '"\nirtu';
+    }
+  }
+  return;
+}
```

Comment:
This is very much an optional, opinionated thing, but I typically favor placing utility functions at the bottom of the file, rather than the top. Since javascript will hoist them anyway, this won't cause any sort of ordering error, and it makes it more clear for others to read the implementation portion of the code, scrolling down to reference the utility functions as needed, then returning to the top of the document to resume the code reading, than scrolling through out-of-context utility functions, then going back up and down to an arbitrary spot in the middle to get back to the reading.